### PR TITLE
Nightly build part 3

### PR DESCRIPTION
**Pull request by sethAmazon**
**Description:** 
Add nightly docker build

**Link to tracking Issue:** 
https://github.com/aws-observability/aws-otel-collector/issues/679

**Testing:** 
Github actions

Requires merge of https://github.com/aws-observability/aws-otel-collector/pull/680 before merging

### CODE REVIEW

File: .github/workflows/nightly-build.yml [hunk 1]
```diff
@@ -23,5 +23,20 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Unit test
-        run: make test
\ No newline at end of file
+      - name: Unit test adot
+        run: make test
+
+      - name: Unit test aws components
```
Comment (sethAmazon) [inline .github/workflows/nightly-build.yml:29]: This will come in part 2

**Comment by dyladan** [comment]
Any ETA on this?

File: Makefile [hunk 2]
```diff
@@ -88,6 +88,12 @@ package-deb: build
 docker-build: 
 	docker build -t $(DOCKER_NAMESPACE)/$(COMPONENT):$(VERSION) -f ./cmd/$(COMPONENT)/Dockerfile .
 
+.PHONY: docker-nightly
+docker-nightly:
+	docker buildx create --use --name mybuildx
+	docker buildx build -t public.ecr.aws/aws-otel-test/nightly-build:latest -f ./cmd/$(COMPONENT)/Dockerfile --platform=linux/amd64 --push .
```
Comment (jefchien) [inline Makefile:94]: Isn't the default docker build for amd64? Why is `buildx` needed? Can we just tag the existing docker build? Like https://github.com/aws-observability/aws-otel-collector/blob/27d3ebf590d08a875fb889c011c7b21886d04fbe/.github/workflows/CD.yml#L344-L352

Comment (sethAmazon) [reply Makefile:94]: Honestly I did this because in the future we may support arm64 since we keep getting requests to add support for it

Comment (sethAmazon) [reply Makefile:94]: This will make it much easier to add that functionality. May be premature optimization.

Comment (jefchien) [reply Makefile:94]: I feel like that would be a bigger change to docker-build itself. What is the difference between the nightly build and the normal docker build?

Comment (sethAmazon) [reply Makefile:94]: Just where it publishes the image

Comment (jefchien) [reply Makefile:94]: So, I think we could do what the CD does and just retag the image and push it to the ECR repo. The buildx changes should be made later to the main docker build itself in a separate PR.

Comment (khanhntd) [reply Makefile:94]: Agree with @jefchien.

**Review by khanhntd — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (khanhntd) [reply Makefile:94]: Agree with @jefchien

File: .github/workflows/nightly-build.yml [hunk 3]
```diff
@@ -24,4 +28,19 @@ jobs:
...
+          aws-access-key-id: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
+          aws-region: us-east-1
+
+      - name: upload to ECR
+        run: |
+          aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/aws-otel-test
+          make amd64-build
+          docker build -t $IMAGE_NAME -f cmd/awscollector/Dockerfile .
+          docker tag $IMAGE_NAME public.ecr.aws/$ECR_REPO:latest
```
Comment (jefchien) [inline .github/workflows/nightly-build.yml:45]: Where is `$ECR_REPO` set?

Comment (sethAmazon) [reply .github/workflows/nightly-build.yml:58]: In env section

Comment (jefchien) [reply .github/workflows/nightly-build.yml:58]: Ah didn't see it the first time

**Pull request merged**